### DISCUSSION TITLE
unable to download bilibili video

### DISCUSSION THREAD

**Discussion by KamiyaMinoru** [InfinityLoop1308/PipePipe#225] [Q&A] [answered]
不管選擇什麼格式, 下載Bilibili的影片都是失敗的
點擊"顯示錯誤"也只會跳出"發生錯誤, 請檢視通知"的popup
在youtube沒有這個問題

**Comment by InfinityLoop1308** [comment]
Could you plz attach error report?

**Reply by KamiyaMinoru** [reply]
![Screenshot_2024-03-02-09-33-20-67_c1a8a2bc1f076b420a754a0676e33b88](https://github.com/InfinityLoop1308/PipePipe/assets/78710607/42046583-e0b8-43eb-835b-3ba441855c72)

點了只給我popup 沒有報告

**Reply by KamiyaMinoru** [reply]
![Screenshot_2024-03-02-09-35-00-49_c1a8a2bc1f076b420a754a0676e33b88](https://github.com/InfinityLoop1308/PipePipe/assets/78710607/3f782fae-29b8-484b-a0f5-ee0e15a3e5b6)

**Reply by KamiyaMinoru** [reply]
is this?

## Exception
* __User Action:__ download post-processing
* __Request:__ https://bilibili.com/BV1mz42197nd?p=1 [ {type=video format=MPEG-4 quality=720P 高清 videoOnly=true}  {type=audio format=m4a bitrate=-1} ]
* __Content Country:__ TW
* __Content Language:__ zh-TW
* __App Language:__ zh_TW
* __Service:__ BiliBili
* __Version:__ 3.3.0
* __OS:__ Linux realme/RMX2144T1/RMX2144L1:11/RKQ1.201217.002/1656816195019:user/release-keys 11 - 30
<details><summary><b>Crash log </b></summary><p>

```
java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.lang.String.substring(String.java:2064)
	at us.shandian.giga.postprocessing.BiliBiliMp4Muxer.process(BiliBiliMp4Muxer.java:25)
	at us.shandian.giga.postprocessing.Postprocessing.run(Postprocessing.java:185)
	at us.shandian.giga.get.DownloadMission.doPostprocessing(DownloadMission.java:706)
	at us.shandian.giga.get.DownloadMission.$r8$lambda$J0GR1urTFneOp2O30GpOM3TNG0I(Unknown Source:0)
	at us.shandian.giga.get.DownloadMission$$ExternalSyntheticLambda0.run(Unknown Source:2)
	at java.lang.Thread.run(Thread.java:923)

```
</details>
<hr>

(I dont know how to recall it but it appear just now)

**Comment by InfinityLoop1308** [comment, marked as answer]
https://github.com/InfinityLoop1308/PipePipe/issues/230
transfer to a issue for clarity. continue the discussion there.

**Comment by InfinityLoop1308** [comment]
May I know your download path?

**Reply by KamiyaMinoru** [reply]
F-droid

**Reply by InfinityLoop1308** [reply]
I mean setting -> Download -> Video Download folder

**Comment by KamiyaMinoru** [comment]
content://com.android.providers.downloads.documents/tree/downloads

InfinityLoop1308 ***@***.***> 於 2024年3月16日 週六 下午2:05寫道：

**Comment by InfinityLoop1308** [comment]
Fixed in 3.3.1

**Reply by KamiyaMinoru** [reply]
Yes! Thank you!! Amazing!!
